test/system/src/test/java/io/pravega/test/system/RetentionTest.java

```diff
@@ -167,14 +163,7 @@ public void retentionTest() throws Exception {
 
         //try reading the event that was written earlier.
         //expectation is it should have been truncated and we should find stream to be empty
-        try {
-            String readEvent = reader.readNextEvent(6000).getEvent();
-            log.debug("Reading event: {} ", readEvent);
-            assertEquals(null, readEvent);
-        }  catch (ReinitializationRequiredException e) {
-            log.error("Unexpected request to reinitialize {}", e);
-            Assert.fail("Unexpected request to reinitialize.Test failed.");
-        }
+        assertThrows(TruncatedDataException.class, () -> reader.readNextEvent(6000));
```

Comment:
i think we should call readNextEvent again and verify that it is able to successfully read the data post truncation too.

Reply:
done.